src/astro/sql/operators/agnostic_load_file.py

```diff
@@ -132,16 +135,66 @@ def _load_dataframe(self, path):
         with smart_open.open(
             path, mode=mode.get(file_type, "r"), transport_params=transport_params
         ) as stream:
-            return deserialiser[file_type](
-                stream, **deserialiser_params.get(file_type, {})
-            )
+            if file_type == "ndjson":
+                df = None
+                count = 0
+                while True:
+                    rows = stream.readlines(DEFAULT_CHUNK_SIZE)
+                    if len(rows) == 0:
+                        break
```

Comment:
This feels off. Maybe you can do something like run `rows = stream.readlines(DEFAULT_CHUNK_SIZE)` before the loop and then say `while len(rows) != 0`